mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -740,6 +742,59 @@ def _get_metric_value_details(self, path, metric):
     def log_metric(self, run_id, metric):
         # simply call _log_metrics and let it handle the rest
         self._log_metrics(run_id, [metric], isSingleMetric=True)
+        self._log_model_metrics(run_id, [metric])
+
+    def _log_model_metrics(
+        self, run_id: str, metrics: list[Metric], path: str = "", is_single_metric: bool = False
+    ) -> None:
+        if not metrics:
+            return
+
+        metric_instances: list[SqlLoggedModelMetric] = []
+        for index, metric in enumerate(metrics):
+            if metric.model_id is None:
+                continue
+            path = path if is_single_metric else append_to_json_path(path, f"[{index}]")
+            metric, value, _is_nan = self._get_metric_value_details(path, metric)
```

Comment:
Can we get rid of `path`? This is a dummy value that seems only to be used for compatibility with `_validate_metric`. IMO it would be clearer to add a second utility method or support `path=False` in the existing utility method.

Reply:
I'll look into https://github.com/mlflow/mlflow/pull/13022 to see what `path` does.

Reply:
Filed https://github.com/mlflow/mlflow/pull/14893.